File: AndroidExample/app/src/main/java/org/bziembanski/androidexample/repository/RemoteRepository.kt
package org.bziembanski.androidexample.repository

import com.jakewharton.retrofit2.converter.kotlinx.serialization.asConverterFactory
import kotlinx.serialization.json.Json
import okhttp3.MediaType
import org.bziembanski.androidexample.data.PeopleResult
import org.bziembanski.androidexample.data.Person
import retrofit2.Retrofit
import retrofit2.http.GET
import retrofit2.http.Path

class RemoteRepository {
    companion object {
        private const val baseUrl = "https://swapi.dev/api/"
    }

    private val retrofit = Retrofit.Builder()
        .addConverterFactory(Json.asConverterFactory(MediaType.get("application/json")))
        .baseUrl(baseUrl)
        .build()

    private val personService = retrofit.create(PersonApiService::class.java)

    suspend fun fetchAll(): List<Person> {
        return personService.fetchAll().results
    }

    suspend fun fetchPerson(personId: Int): Person {
        return personService.fetchPerson(personId)
    }
}

interface PersonApiService {
    @GET("people")
    suspend fun fetchAll(): PeopleResult

    @GET("people/{id}")
    suspend fun fetchPerson(@Path("id") personId: Int): Person
}

File: ReactNativeExample/repository/remote_repository.ts
import { Int32 } from "react-native/Libraries/Types/CodegenTypes";
import { PeopleResult, Person } from "../data/person";
import axios from "axios";


export class RemoteRepository{
    instance = axios.create({
        baseURL: 'https://swapi.dev/api/',
    });
    async fetchAll(): Promise<Person[]>{
        let response =  await this.instance.get("people");
        let result: PeopleResult = response.data;
        return result.results;
    }

    async fetchPerson(personId: Int32): Promise<Person>{
        let response =  await this.instance.get(`people/${personId}`);
        let result: Person = response.data;
        return result;
    }
}